ground/src/main/java/com/google/android/ground/ui/common/NavigationRequest.kt

```diff
@@ -17,10 +17,10 @@ package com.google.android.ground.ui.common
 
 import androidx.navigation.NavDirections
 
-sealed interface NavigationRequest
+sealed interface NavigationRequest {
+  data object NavigateUp : NavigationRequest
 
-data class NavigateTo(val directions: NavDirections) : NavigationRequest
+  data object FinishApp : NavigationRequest
```

Comment:
What is the rational behind changing it from `object` to `data object`? other than `data object` gives you the power to have a toString() function as written in the PR description, which just prints the name of the object

looks like an extra work we are trying to give to compiler to generate some functions for us, which we are currently not using.

WDYT? 
Let me know if I am missing anything here.

Reply:
This also adds more boilerplate it seems in referencing code.

Reply:
Thanks for the comment. 😀

@anandwana001 
The functions generated by the `data object`, such as `toString()`, may not be used in production code, but they can be useful for debugging during development. This can be easily understood through a [link](https://kotlinlang.org/docs/whatsnew1720.html#improved-string-representations-for-singletons-and-sealed-class-hierarchies-with-data-objects).
Also, Requesting additional work from the compiler is also justified because the Kotlin compiler is optimized to handle methods like `toString`, `equals`, and `hashCode` (e.g., for data classes).

@gino-m 
I understand that this leads to some additional boilerplate, but I think it has improved readability.

Reply:
Got the idea now, cool, let's have it.